app/src/main/java/com/dao/mydebts/adapters/DebtsAdapter.java

```diff
@@ -174,6 +174,32 @@ public void clear() {
 
     private class DebtApproveListener implements View.OnClickListener {
 
+        private final AlertDialog dialog;
+        private Debt toApprove;
+
+        public DebtApproveListener() {
+            AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+            builder.setTitle(R.string.approve_dialog_title);
+            builder.setMessage(R.string.approve_dialog_message);
+            builder.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int id) {
+                    // send back to activity
+                    DebtsListActivity casted = (DebtsListActivity) mContext;
+                    Handler bHandler = casted.getBackgroundHandler();
+                    Message msg = Message.obtain(bHandler, DebtsListActivity.MSG_APPROVE_DEBT, toApprove);
+                    bHandler.sendMessage(msg);
+                    toApprove = null;
+                    dialog.dismiss();
+                }
+            });
+            builder.setNegativeButton(android.R.string.cancel, new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int id) {
+                    dialog.dismiss();
```

Comment:
This one will be initialized once for every holder item, better make it lazy-init with just `onClick` method. This will also eliminate possibility of invoking `show` on already dismissed dialog resulting in a crash.